vmtkScripts/contrib/vmtknumpytocenterlines.py

```diff
@@ -0,0 +1,111 @@
+##       University at Buffalo
+##       The Jacobs Institute
+
+from __future__ import absolute_import  # NEEDS TO STAY AS TOP LEVEL MODULE FOR Py2-3 COMPATIBILITY
+import vtk
+import sys
+
+from vmtk import vtkvmtk
+from vmtk import vmtkrenderer
+from vmtk import pypes
+
+try:
+    import numpy as np
+except ImportError:
+    raise ImportError('Unable to Import vmtknumpytosurface module, numpy is not installed')
+
+class vmtkNumpyToCenterlines(pypes.pypeScript):
+    def __init__(self):
+
+        pypes.pypeScript.__init__(self)
+
+        self.ArrayDict = None
+
+        self.SetScriptName('vmtkNumpyToCenterlines')
+        self.SetScriptDoc('Takes a nested python dictionary containg numpy arrays specifying vertex '
+                          'points, associated scalar data, and cell data specifying line connectivity'
+                          'and returns a VTK centerlines vtkPolyData file')
+        self.SetInputMembers([
+            ['ArrayDict', 'i', 'dict', 1, '', 'the input dictionary']])
+        self.SetOutputMembers([
+            ['Centerlines', 'o', 'vtkPolyData', 1, '', 'the output surface', 'vmtksurfacewriter']])
+
+    def Execute(self):
+
+        polyData = vtk.vtkPolyData()
+
+        points = vtk.vtkPoints()
+        for xyzPoint in self.ArrayDict['Points']:
+            points.InsertNextPoint(xyzPoint)
+
+        polyData.SetPoints(points)
+
+        self.PrintLog('converting numpy array to centerlines')
+        pointDataKeys = self.ArrayDict['PointData'].keys()
+        for key in pointDataKeys:
+
+            pointDataDType = str(self.ArrayDict['PointData'][key].dtype)
+            if pointDataDType.find('float') != -1:
+                pointDataArray = vtk.vtkFloatArray()
+            if pointDataDType.find('int') != -1:
+                pointDataArray = vtk.vtkIntArray()
+
+            try:
+                pointDataComponents = self.ArrayDict['PointData'][key].shape[1]
+            except IndexError:
+                pointDataComponents = 1
+
+            pointDataArray.SetNumberOfComponents(pointDataComponents)
+            pointDataArray.SetName(key)
+
+            if pointDataComponents == 1:
+                for pointData in self.ArrayDict['PointData'][key]:
+                    pointDataArray.InsertNextValue(pointData)
+                polyData.GetPointData().SetActiveScalars(key)
+                polyData.GetPointData().SetScalars(pointDataArray)
+            else:
+                for pointData in self.ArrayDict['PointData'][key]:
+                    pointDataArray.InsertNextTuple(pointData)
+                polyData.GetPointData().SetActiveVectors(key)
+                polyData.GetPointData().SetVectors(pointDataArray)
+
+        cellDataArray = vtk.vtkCellArray()
+        numberOfCells = len(self.ArrayDict['CellData']['CellPointIds'].keys())
+        for cellId in range(numberOfCells):
+            numberOfCellPoints = self.ArrayDict['CellData']['CellPointIds'][str(cellId)].size
```

Comment:
I don't fully understand why you're required to use a string `cellId`. Does this come from the vtk util function that generates the dictionary?